libs/jsonschema/from_type_test.go

```diff
@@ -0,0 +1,462 @@
+package jsonschema
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/databricks/cli/libs/jsonschema/test_types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFromTypeBasic(t *testing.T) {
+	type myStruct struct {
+		S string      `json:"s"`
+		I *int        `json:"i,omitempty"`
+		V interface{} `json:"v,omitempty"`
+
+		// These fields should be ignored in the resulting schema.
+		NotAnnotated     string
+		DashedTag        string `json:"-"`
+		InternalTagged   string `json:"internal_tagged" bundle:"internal"`
+		DeprecatedTagged string `json:"deprecated_tagged" bundle:"deprecated"`
+		ReadOnlyTagged   string `json:"readonly_tagged" bundle:"readonly"`
+	}
+
+	strRef := "#/$defs/string"
+	boolRef := "#/$defs/bool"
+	intRef := "#/$defs/int"
+	interfaceRef := "#/$defs/interface"
+
+	tcases := []struct {
+		name     string
+		typ      reflect.Type
+		expected Schema
+	}{
+		{
+			name: "int",
+			typ:  reflect.TypeOf(int(0)),
+			expected: Schema{
+				Type: "integer",
+			},
+		},
+		{
+			name: "string",
+			typ:  reflect.TypeOf(""),
```

Comment:
Nit: can you `str("")` for symmetry with the others/clarity?

Reply:
done